pkg/util/utils_test.go

```diff
@@ -0,0 +1,151 @@
+	matchString := "volume"
+
+	ret := ListContains(elements, matchString)
+	assert.Equal(t, ret, true, "%s entity not found in the list", matchString)
+}
+
+/*
+To test negative case of utils.ListContains function.
+Test if given element is not present in the list.
+Asserts if element is found in the list.
+*/
+func TestListContainsNoElement(t *testing.T) {
+	// list containg elements
+	elements := []string{"node", "drive", "volume"}
+	matchString := "portworx"
+
+	ret := ListContains(elements, matchString)
+	assert.Equal(t, ret, false, "%s entity found in the list", matchString)
+}
+
+/*
+To test positive case of utils.ListHaveMatch.
+Test if given element present in the both the list.
+Assert if none of the elements is not found in both the list.
+*/
+func TestListHaveMatchPresent(t *testing.T) {
+	elements := []string{"node", "drive", "volume", "portworx"}
+	match := []string{"portworx", "osd"}
+
+	ret := ListHaveMatch(elements, match)
+	assert.Equal(t, ret, true, "Elements %s not found in the list", match)
+}
+
+/*
+Test for negative case of utils.ListHaveMatch.
+Tests if the given entity is not present in the list.
+Assert if any one of the elements is found in both the list.
+*/
+func TestListHaveMatchNotPresent(t *testing.T) {
+	elements := []string{"node", "drive", "volume", "portworx"}
+	match := []string{"oci", "osd"}
+
+	ret := ListHaveMatch(elements, match)
+	assert.Equal(t, ret, false, "One of the elements %s found in the list", match)
+}
+
+/*
+Test for positive case of utils.StringMapToCommaString
+Test if the given map can be converted to valid string.
+Assert on conversion failure.
+*/
+func TestStringMapToCommaString(t *testing.T) {
+	elements := map[string]string{
+		"pod":     "portworx",
+		"cluster": "k8s",
+	}
+	expectedResult := "pod=portworx,cluster=k8s"
+	ret := StringMapToCommaString(elements)
+	assert.Equal(t, ret, expectedResult, "Failed to convert (k,v) to string")
+}
+
+/*
+Test for positive case of utils.CommaStringToStringMap
+Tests if the given valid string is converted to  (k,v) pair.
+Asserts if the conversion fails
+*/
+func TestCommaStringToStringMapPositive(t *testing.T) {
+	element := "pod=portworx,cluster=k8s"
+	expectedResult := map[string]string{
+		"pod":     "portworx",
+		"cluster": "k8s",
+	}
+
+	ret, _ := CommaStringToStringMap(element)
+	state := reflect.DeepEqual(ret, expectedResult)
+	assert.Equal(t, state, true, "Failed to convert string %s to (k,v) pair", element)
+}
+
+/*
+Test function for negative cases of utils.CommaStringToStringMap
+Tests if the given valid string is not converted to  (k,v) pair.
+Asserts if the conversion succeeds.
+*/
+func TestCommaStringToStringMapNegative(t *testing.T) {
+	// case 1
+	element := "pod+portworx,cluster/k8s"
+	expectedResult := map[string]string{
+		"pod":     "portworx",
+		"cluster": "k8s",
+	}
+
+	state := deepCompare(element, expectedResult)
+	assert.Equal(t, state, false, "Successfully converted string %s to (k,v) pair "+
```

Comment:
Suggestion: No need to add messages. Only add messages during debug.

Reply:
Done. Removed messages from all assert statements.